web/web/api/v1/sns.py

```diff
@@ -0,0 +1,30 @@
+from flask import Flask, request, Blueprint
+import json
+
+sns_api_bp =  Blueprint('sns_api_bp', __name__)
+
+
+@sns_api_bp.route('/sns', methods = ['POST'])
+def subscriber_information():
+    # AWS sends JSON with text/plain mimetype
+    try:
+        js = json.loads(request.data)
+    except:
+        return 'An error occuered'
+    hdr = request.headers.get('x-amz-sns-message-type')
+    # subscribe to the SNS topic
+    if hdr == 'Notification':
+        parse_real_power(js['Message'])
+    elif hdr == 'SubscriptionConfirmation' and 'SubscribeURL' in js:
+        return js['SubscribeURL']
+    return 'OK'
+
+
+def parse_real_power(message):
```

Comment:
Can you share an example of what this message looks like? From my understanding this should be json already....I'm wondering if it's double encoded out of curiosity.

Reply:
b'{"DeviceID": "TEC_GISMO", "DeviceInformation": [{"sunnyboy_inverter.power_l1": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.reactive_power_l2": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.calc_ac_power_kw": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.reactive_power_l1": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.dc_voltage_input_1": {"timestamp": 1606332463830, "value": 394.07}, "sunnyboy_inverter.dc_voltage_input_2": {"timestamp": 1606332463830, "value": 395.38}, "sunnyboy_inverter.grid_voltage_phase_l2": {"timestamp": 1606332463830, "value": 117.26}, "sunnyboy_inverter.apparent_power_l1": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.apparent_power_l2": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.grid_voltage_phase_l1": {"timestamp": 1606332463830, "value": 117.14}, "sunnyboy_inverter.grid_frequency": {"timestamp": 1606332463830, "value": 60.02}, "sunnyboy_inverter.power_l2": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.dc_current_input_1": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.dc_current_input_2": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.dc_power_input_2": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.dc_power_input_1": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.grid_current_phase_l1": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.grid_current": {"timestamp": 1606332463830, "value": 0.0}, "sunnyboy_inverter.gridms_phv_a2b": {"timestamp": 1606332463830, "value": 234.4}, "sunnyboy_inverter.grid_current_phase_l2": {"timestamp": 1606332463830, "value": 0.0}}]}'